### PR TITLE
Fix handling of relative-file storage and auto linking

### PR DESCRIPTION
**Pull request by koppor**
Fixes https://github.com/JabRef/jabref-issue-melting-pot/issues/479

Follow-up to https://github.com/JabRef/jabref/pull/9113

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/model/database/BibDatabaseContext.java [hunk 1]
```diff
@@ -154,40 +153,45 @@ public boolean isStudy() {
...
                 if (parentPath == null) {
                     parentPath = Path.of(System.getProperty("user.dir"));
+                    LOGGER.warn("Parent path of database file {} is null. Falling back to {}.", dbPath, parentPath);
                 }
                 Objects.requireNonNull(parentPath, "BibTeX database parent path is null");
-                fileDirs.add(parentPath);
+                Path absolutePath = parentPath.toAbsolutePath();
+                if (!fileDirs.contains(absolutePath)) {
+                    fileDirs.add(absolutePath);
+                }
```
Comment (calixtus) [inline src/main/java/org/jabref/model/database/BibDatabaseContext.java:186]: I believe we can make a set out of this...

**Review by Siedlerchr — DISMISSED** [review]

File: src/main/java/org/jabref/model/database/BibDatabaseContext.java (hunk 1 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/model/database/BibDatabaseContext.java:186]: Not useful, we need to keep the order and we only have like 3 possible values

Comment (calixtus) [reply src/main/java/org/jabref/model/database/BibDatabaseContext.java:186]: sortedset

Comment (koppor) [reply src/main/java/org/jabref/model/database/BibDatabaseContext.java:186]: > sortedset

We need to keep the order.

Sequenced Collection is the "right" data type. It checks for containment at each insert. With my approach, only one check needs to be made.

Moreover, we have at most two elements to check. We use an ArrayList where the "forEach" is faster than with a LinkedList.

Changing the return type to a sequencedcollection results in much code changes. Using it internally and then converting to a list results in more CPU cycles than "just" checking two elements.

I can rewrite to return SequencedCollection, write an ADR, add Java comments, link to this text or just leave it.

What should I do?

Comment (koppor) [reply src/main/java/org/jabref/model/database/BibDatabaseContext.java]: OK, I just switched to `LinkedHashSet`. This also ensures that the second path is checked for uniqueness.

**Review by Siedlerchr — APPROVED** [review]

File: src/main/java/org/jabref/model/database/BibDatabaseContext.java (hunk 1 above)
Comment (calixtus) [reply src/main/java/org/jabref/model/database/BibDatabaseContext.java]: Was just an idea, the code just smelled like a set. no strong opinion on this.

**Review by calixtus — APPROVED** [review]

**Pull request merged**